Allow user to define periodic event type

azadon opened the pull request:
Extending add_periodic with optional argument for configuring event behavior after creation - SD_EVENT_ON, SD_EVENT_OFF or SD_EVENT_ONESHOT. This allows user to create a periodic event which can stop itself from event's callback function.

Withing PR I also added function for getting information if event is enabled or not and description contains also this information.

azadon commented on src/evl.c:
```diff
@@ -10,11 +10,18 @@
+
 	ret = sd_event_source_get_description(evsrc, &desc);
-	lua_pushfstring(L, "event_source [%s] %p", (ret<0)?"unkown":desc, evsrc);
+	lua_pushfstring(L, "event_source [%s, %s] %p", (ret<0)?"unkown":desc, (enabled==-1?"oneshot":(enabled==1?"enabled":"disabled")), evsrc);
```
I am not proud of this, but ... I wanted to keep a oneliner :grimacing:

kmarkus replied:
that's fine :+1:

kmarkus commented on src/evl.c:
```diff
@@ -303,22 +303,30 @@ static int timer_callback(sd_event_source *evsrc, uint64_t usec, void* userdata)
 	luaL_checktype(L, 4, LUA_TFUNCTION);
+	enabled = luaL_optinteger(L, 5, SD_EVENT_ON);
+	if (lua_isnone(L, 5) == 0) {
+		// when there was anything pop it, because stack is used further down.
```
Please switch to C-style comments (`/* ... */`)

kmarkus reviewed (approved):
Please update the README with the changed API and add a short entry to the `ChangeLog` section. 

Thank you very much!

azadon commented:
I've updated README, while doing it I realized that `get_enabled` should return the same type which `set_enabled` accepts. Also, I changed default event mode to ONESHOT, as it makes more sense.

kmarkus commented:
> I've updated README, while doing it I realized that `get_enabled` should return the same type which `set_enabled` accepts. Also, I changed default event mode to ONESHOT, as it makes more sense.

Thank you for the update. But I don't think we should change the default to ONESHOT, it seems counterintuitive for an API that is called `add_periodic`. Also it will probably break a lot of existing uses.

kmarkus commented:
> > I've updated README, while doing it I realized that `get_enabled` should return the same type which `set_enabled` accepts. Also, I changed default event mode to ONESHOT, as it makes more sense.
> 
> Thank you for the update. But I don't think we should change the default to ONESHOT, it seems counterintuitive for an API that is called `add_periodic`. Also it will probably break a lot of existing usesA

I think a cleaner way forward would be to introduce a new `_add_time` API that more faithfully covers the main use-cases of SD_EVENT_ADD_TIME(3), including changing expiration time. `add_periodic` could then become a convenience API that uses this internally.
